### DISCUSSION TITLE
Is the change from immutable ARRAY type to mutable considered as a change of type?

### DISCUSSION THREAD

**Discussion by razvan286** [sqlalchemy/sqlalchemy#9139] [Usage Questions] [answered]
I use sqlalchemy 1.4.46 and alembic for migrations.

Previously, my column type looked like this:

`has_bubble_in_countries = sa.Column((ARRAY(sa.Enum(Country))), nullable=False, default=[])`

Which was not allowing me to add or remove any elements from this array.

Then, I made it immutable by changing it like this:

`has_bubble_in_countries = sa.Column(MutableList.as_mutable(ARRAY(sa.Enum(Country))), nullable=False, default=[])`

Does this change require a migration?

My first thought was that this is not an altering of the type so I considered that the migration is not needed.

**Comment by zzzeek** [comment, marked as answer]
this is only a Python behavioral change and not a change in the schema of the database, since the datatype is ARRAY that isn't changing, so no migration is needed.
